lib/languages/js/source-code/source-code.js

```diff
@@ -868,6 +868,40 @@ class SourceCode extends TokenStore {
 		return ancestorsStartingAtParent.reverse();
 	}
 
+	/**
+	 * Determines whether the given identifier node is a reference to a global variable.
+	 * @param {ASTNode} node `Identifier` node to check.
+	 * @returns {boolean} True if the identifier is a reference to a global variable.
+	 */
+	isGlobalReference(node) {
+		if (!node) {
+			throw new TypeError("Missing required argument: node.");
+		}
+
+		if (node.type !== "Identifier") {
+			return false;
+		}
+
+		const variable = this.scopeManager.scopes[0].set.get(node.name);
+
+		if (!variable) {
+			return false;
+		}
+
+		const references = variable.references;
+
+		for (const reference of references) {
+			if (
+				reference.identifier === node &&
+				reference.resolved.scope.type === "global"
+			) {
```

Comment:
```suggestion
			if (reference.identifier === node) {
```

`reference.resolved` is `variable`, which is already certainly in the global scope.